File: level1/02-palindrome/main.go
package main

import (
	"fmt"
	"strings"
)

func reverse(s string) string {
	sArray := strings.Split(s, "")
	len := len(s)
	mid := len / 2
	for i := 0; i < mid; i++ {
		sArray[i], sArray[len-1-i] = sArray[len-1-i], sArray[i]
	}

	return strings.Join(sArray, "")
}

func palindrome(s string) bool {
	return s == reverse(s)
}

func main() {
	fmt.Println("Is Palindrome abba:", palindrome("abba"))
	fmt.Println("Is Palindrome babba:", palindrome("babba"))
}

File: level1/02-palindrome/main.rb
#!/usr/bin/env ruby

def palindrome(str)
  return  ( "#{str}" == "#{reverse(str)}" ) 
end



def reverse(str)
  
  len = str.size
  
  if len > 0
    
    (0 ... len/2).to_a.reverse_each do |i|
      str[i], str[len-1-i] =  str[len-1-i] , str[i]     
    end

    return str
  end
  
  ""
end

p palindrome("abba")
p palindrome("asdf")